wire/protobuf/serializer.go

```
@@ -0,0 +1,182 @@
+}
+
+// Decode decodes an envelope from the reader, that was encoded using protocol
+// buffers serialization format.
+func (serializer) Decode(r io.Reader) (env *wire.Envelope, err error) { //nolint: funlen, cyclop
+	env = &wire.Envelope{}
+
+	protoEnv, err := readEnvelope(r)
+	if err != nil {
+		return nil, err
+	}
+
+	env.Sender, env.Recipient, err = unmarshalSenderRecipient(protoEnv)
+	if err != nil {
+		return nil, err
+	}
+
+	switch protoMsg := protoEnv.Msg.(type) {
+	case *Envelope_PingMsg:
+		env.Msg = toPingMsg(protoMsg)
+	case *Envelope_PongMsg:
+		env.Msg = toPongMsg(protoMsg)
+	case *Envelope_ShutdownMsg:
+		env.Msg = toShutdownMsg(protoMsg)
+	case *Envelope_AuthResponseMsg:
+		env.Msg = &wire.AuthResponseMsg{}
+	case *Envelope_LedgerChannelProposalMsg:
+		env.Msg, err = toLedgerChannelProposalMsg(protoMsg)
+	case *Envelope_SubChannelProposalMsg:
+		env.Msg, err = toSubChannelProposalMsg(protoMsg)
+	case *Envelope_VirtualChannelProposalMsg:
+		env.Msg, err = toVirtualChannelProposalMsg(protoMsg)
+	case *Envelope_LedgerChannelProposalAccMsg:
+		env.Msg, err = toLedgerChannelProposalAccMsg(protoMsg)
+	case *Envelope_SubChannelProposalAccMsg:
+		env.Msg = toSubChannelProposalAccMsg(protoMsg)
+	case *Envelope_VirtualChannelProposalAccMsg:
+		env.Msg, err = toVirtualChannelProposalAccMsg(protoMsg)
+	case *Envelope_ChannelProposalRejMsg:
+		env.Msg = toChannelProposalRejMsg(protoMsg)
+	case *Envelope_ChannelUpdateMsg:
+		env.Msg, err = toChannelUpdateMsg(protoMsg)
+	case *Envelope_VirtualChannelFundingProposalMsg:
+		env.Msg, err = toVirtualChannelFundingProposalMsg(protoMsg)
+	case *Envelope_VirtualChannelSettlementProposalMsg:
+		env.Msg, err = toVirtualChannelSettlementProposalMsg(protoMsg)
+	case *Envelope_ChannelUpdateAccMsg:
+		env.Msg = toChannelUpdateAccMsg(protoMsg)
+	case *Envelope_ChannelUpdateRejMsg:
+		env.Msg = toChannelUpdateRejMsg(protoMsg)
+	case *Envelope_ChannelSyncMsg:
+		env.Msg, err = toChannelSyncMsg(protoMsg)
```

Please throw an error in the default case.

---

Add in this commit: d995e181.